packages/muelu/test/unit_tests/MueLu_Test_ETI.hpp

```diff
@@ -95,6 +100,18 @@ bool Automatic_Test_ETI(int argc, char *argv[]) {
   Kokkos::initialize(argc, argv);
 #endif
 
+  // Create handles for cuBLAS and cuSPARSE. Otherwise they get
+  // created on the first call to these libraries, and that can mess
+  // up timings.
+#ifdef HAVE_MUELU_KOKKOSKERNELS
+  KokkosKernels::Experimental::Controls controls;
+# ifdef KOKKOSKERNELS_ENABLE_TPL_CUBLAS
+  controls.getCublasHandle();
+# endif
+# ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+  controls.getCusparseHandle();
+# endif
+#endif
```

Comment:
You could consider adding a fence here for good measure, I believe that cuSPARSE is doing some async memory touches in the handle.

Reply:
Ok, will do. In my runs it seemed to not matter, but maybe I'm hitting an early fence somewhere else..